xarray/plot/utils.py

```diff
@@ -1268,3 +1293,325 @@ def _parse_size(data, norm):
     sizes = dict(zip(levels, widths))
 
     return pd.Series(sizes)
+
+
+class _Normalize(Sequence):
```

Comment:
I wrote this for better or worse... 
I had some idea a while ago that maybe it is possible to use matplotlibs normalize classes instead of this but I'm not sure if it would become more readable.